### ISSUE TITLE
d3.js:1378 Error: <circle> attribute cx: Expected length, "NaN".

### ISSUE THREAD

**Issue by rtm** [marmelab/EventDrops#149]
I worked around the error I reported earlier related to npm install by cloning the repo and linking it into my project. That is working fine.

I have a trivially simple example that looks like this:

```
import * as d3 from 'd3';
import eventDrops from 'event-drops/dist/eventDrops';

    const chart = eventDrops()
      .start(new Date(2017, 1, 1))
      .end(new Date(2017, 4, 31))
      .eventColor("red")
      .margin({ top: 60, left: 0, bottom: 40, right: 50 })
      .eventLineColor("blue")
      .labelsWidth(100)
    ;

    d3.select("#eventDrops")
      .datum([
        {name: "Bob", data: [new Date(), new Date(2017, 3, 18)]},
        {name: "Sally", data: [new Date(2017, 4, 3), new Date(2017, 5, 22)]},
      ])
      .call(chart);
```

The chart displays, but whenever I drag the timeline, I get

> d3.js:1378 Error: <circle> attribute cx: Expected length, "NaN".

And the drops disappear.

I will continue to try to isolate this problem, and/or create a simple reproducible test case (at the moment, this is inside my large Angular app), but is there any obvious reason this error would be appearing?

**Comment by rtm** [comment]
The issue seems to be in this line in `zoom.js`:

    return scalingFunction(new Date(d.date));

This should be calling the `date` extraction function, instead of hardwiring the `date` property.

**Comment by jpetitcolas** [comment]
Will be fixed by #133. Thanks for your investigation. :)

**Comment by BartoszJanota** [comment]
Hi, I'm trying to use your example and I'm getting this issue again. Was it solved? @jpetitcolas
